src/Rings/mpoly-graded.jl

```diff
@@ -1562,6 +1562,397 @@ end
 ### Homogenization and Dehomogenization
 ############################################################################
 
+## 2024-02-06  New UI to homogenization & dehomogenization
+## used partly the code from 2023-07 (see below)
+
+
+struct homogenizer
+  P::MPolyRing              # orignal poly ring (?not graded?)
+  P_homog::MPolyDecRing     # graded poly ring: same vars as P, plus g extra homogenizing vars
+  VarMap::Vector{Int}       # var[k] in P embeds to var[VarMap[k]] in P_homog
```

Comment:
Did you consider to store and use an actual ring map in the internals? Is there a reason not to do so?

Reply:
What is a ring map?  Maybe this can be done after the book deadline (since I don't think it will change the UI).